src/databricks/labs/lakebridge/install.py

```diff
@@ -447,123 +93,36 @@ def run(
         logger.info("Installation completed successfully! Please refer to the documentation for the next steps.")
         return config
 
-    def has_installed_transpilers(self) -> bool:
-        """Detect whether there are transpilers currently installed."""
+    def upgrade_installed_transpilers(self) -> bool:
+        """Detect and upgrade, if possible and necessary, installed transpilers."""
         installed_transpilers = self._transpiler_repository.all_transpiler_names()
         if installed_transpilers:
             logger.info(f"Detected installed transpilers: {sorted(installed_transpilers)}")
-        return bool(installed_transpilers)
-
-    def install_bladebridge(self, artifact: Path | None = None) -> None:
-        local_name = "bladebridge"
-        pypi_name = "databricks-bb-plugin"
-        wheel_installer = WheelInstaller(self._transpiler_repository, local_name, pypi_name, artifact)
-        wheel_installer.install()
-
-    def install_morpheus(self, artifact: Path | None = None) -> None:
-        if not self.is_java_version_okay():
-            logger.error(
-                "The morpheus transpiler requires Java 11 or above. Please install Java and re-run 'install-transpile'."
-            )
-            return
-        product_name = "databricks-morph-plugin"
-        group_id = "com.databricks.labs"
-        artifact_id = product_name
-        maven_installer = MavenInstaller(self._transpiler_repository, product_name, group_id, artifact_id, artifact)
-        maven_installer.install()
-
-    @classmethod
-    def is_java_version_okay(cls) -> bool:
-        detected_java = cls.find_java()
-        match detected_java:
-            case None:
-                logger.warning("No Java executable found in the system PATH.")
-                return False
-            case (java_executable, None):
-                logger.warning(f"Java found, but could not determine the version: {java_executable}.")
-                return False
-            case (java_executable, bytes(raw_version)):
-                logger.warning(f"Java found ({java_executable}), but could not parse the version:\n{raw_version}")
-                return False
-            case (java_executable, tuple(old_version)) if old_version < (11, 0, 0, 0):
-                version_str = ".".join(str(v) for v in old_version)
-                logger.warning(f"Java found ({java_executable}), but version {version_str} is too old.")
-                return False
-            case _:
-                return True
-
-    def install_artifact(self, artifact: str):
+        upgraded = False
+        for transpiler_installer in self._transpiler_installers:
+            name = transpiler_installer.name
+            if name in installed_transpilers:
+                logger.info(f"Checking for {name} upgrades...")
+                upgraded |= transpiler_installer.install()
+        # If we upgraded anything, the configuration process needs to run again.
```

Comment:
I like this nudge

Reply:
It's more than a nudge, it's also necessary because after an update the previous configuration might not be valid any more (eg. dialect removed), so this forces it into a valid state.